### PR TITLE
Adding option to model type in OD

### PR DESCRIPTION
**Pull request by shreyajain17**

### CODE REVIEW

File: src/toolkits/object_detection/object_detector.cpp [hunk 1]
```diff
@@ -281,6 +281,13 @@ void object_detector::init_options(
       "If True, print progress updates and model details.",
       true,
       true);
+  options.create_categorical_option(
+     /* name              */ "model",
+      /* description       */
+      "Defines the model type",
```
Comment (TobyRoseman) [inline src/toolkits/object_detection/object_detector.cpp]: FYI - Looks like there are some indentation issues here. I also don't think you want a new line here.

Comment (shreyajain17) [reply src/toolkits/object_detection/object_detector.cpp]: Yeah. I just did it.

Comment (shreyajain17) [reply src/toolkits/object_detection/object_detector.cpp]: I just did it.

File: src/toolkits/object_detection/object_detector.cpp [hunk 2]
```diff
@@ -281,6 +281,14 @@ void object_detector::init_options(
       "If True, print progress updates and model details.",
       true,
       true);
+  options.create_categorical_option(
+      /* name              */ "model",
+      /* description       */
+      "Defines the model type",
+      /* default_value     */ "darknet-yolo",
+      /* allowed_values    */
+      {flexible_type("darknet-yolo"), flexible_type("other")},
```
Comment (nickjong) [inline src/toolkits/object_detection/object_detector.cpp]: If we don't know yet what the other categorical values will be, for now we can just make this an arbitrary string value

Comment (shreyajain17) [reply src/toolkits/object_detection/object_detector.cpp]: Done

**Pull request merged**